core/src/main/kotlin/org/evomaster/core/search/service/mutator/geneMutation/MutationBoundaryUpdate.kt

```diff
@@ -0,0 +1,46 @@
+package org.evomaster.core.search.service.mutator.geneMutation
+
+/**
+ * created by manzh on 2019-09-19
+ */
+abstract class MutationBoundaryUpdate<T> (var preferMin : T, var preferMax : T, var counter : Int = 0, var reached : Boolean = false) where T : Number{
+
+    abstract fun copy() : MutationBoundaryUpdate<T>
+
+    fun setCounter(improved : Boolean){
```

Comment:
methods that starts with `set` and `get` have very special meaning in Kotlin (eg, they will create properties). as here you are not setting the counter, should use a different name, eg something like `updateCounter`